File: src/jvmception/objects/cp/CpClass.java
package jvmception.objects.cp;

import java.io.DataInputStream;
import java.io.IOException;

import jvmception.jvmtypes.IUnitSerializable;
import jvmception.jvmtypes.JVMReference;

public class CpClass extends CpInfo {
	private int nameIndex;
	
	public int getNameIndex() {
		return this.nameIndex;
	}
	
	public CpClass(DataInputStream stream) throws IOException {
		nameIndex = stream.readUnsignedShort();
	}

	@Override
	public IUnitSerializable toIUnitSerializable() {
		return JVMReference.NULL_REFERENCE;
	}
}

File: src/jvmception/objects/cp/CpClass.cs

using System.IO;

public class CpClass : CpInfo
{
    private int nameIndex;

    public int getNameIndex()
    {
        return this.nameIndex;
    }

    public CpClass(BinaryReaderBigEndian stream)
    {
        nameIndex = stream.ReadUInt16();
    }

    public override IUnitSerializable toIUnitSerializable()
    {
        return JVMReference.NULL_REFERENCE;
    }
}
